### PR TITLE
docs(config): add module definitions

### PR DESCRIPTION
**Pull request by qbzzt**

### CODE REVIEW

File: docs/pages/config.mdx [hunk 1]
```diff
@@ -48,6 +56,23 @@ export default defineWorld({
...
+      args: [
+        {
+          type: "bytes32",
+          value: resourceToHex({
+            type: "table",
+            namespace: "app",
+            name: "Tasks",
+          }),
+        },
+      ],
```
Comment (holic) [inline docs/pages/config.mdx:73]: can we make the default recommendation use `resolveTableId` since it's less complex? and save this for the more advanced usage?
```suggestion
      args: [resolveTableId("Tasks")],
```

File: docs/pages/config.mdx [hunk 2]
```diff
@@ -17,6 +17,14 @@ The is an example of a `World` config:
 ```tsx
 import { defineWorld } from "@latticexyz/world";
 
+// Used for the module parameters, not required otherwise.
+// If you use this, make sure to run:
+//
+// pnpm install @latticexyz/common
+//
+// In the packages/contracts directory.
+import { resourceToHex } from "@latticexyz/common";
```
Comment (ludns) [inline docs/pages/config.mdx:26]: this is importing the wrong thing right? seems like you use `resolveTableId`, not `resourceToHex`

Comment (qbzzt) [reply docs/pages/config.mdx]: Yes, I should have fixed it when I changed it below. Thank you for catching this - fixed.

File: docs/pages/config.mdx [hunk 3]
```diff
@@ -125,3 +130,78 @@ The global configuration keys are all optional.
...
+
+       ```sh
+       cd packages/contracts
+       pnpm install @latticexyz/config
+       ```
+
+    1. In `mud.config.ts`, import `resolveTableId`.
+
+       ```typescript
+       import { resolveTableId } from "@latticexyz/config/register";
```
Comment (holic) [inline docs/pages/config.mdx:163]:
```suggestion
       import { resolveTableId } from "@latticexyz/world/internal";
```

File: docs/pages/config.mdx [hunk 4]
```diff
@@ -17,6 +17,14 @@ The is an example of a `World` config:
 ```tsx
 import { defineWorld } from "@latticexyz/world";
 
+// Used for the module parameters, not required otherwise.
+// If you use this, make sure to run:
+//
+// pnpm install @latticexyz/common
+//
+// In the packages/contracts directory.
+import { resolveTableId } from "@latticexyz/config/register";
```
Comment (holic) [inline docs/pages/config.mdx:26]: like my previous comment, this should be imported from world (see https://mud.dev/changelog for details)

File: docs/pages/config.mdx [hunk 5]
```diff
@@ -125,3 +123,64 @@ The global configuration keys are all optional.
...
+    </details>
+
+    <details>
+
+    <summary>Use `resourceToHex` to specify the full resource information.</summary>
+
+    1. In `mud.config.ts`, import `resourceToHex`.
+
+       ```typescript
+       import { resourceToHex } from "@latticexyz/world/internal";
```
Comment (holic) [inline docs/pages/config.mdx:168]: this still lives in common

and I'm not sure we need to document this as I'd rather folks not lean on this particular approach (gives us less flexibility to change the underlying implementation)

Comment (qbzzt) [reply docs/pages/config.mdx:168]: OK, I'll remove that.

File: docs/pages/config.mdx [hunk 6]
```diff
@@ -12,18 +12,19 @@ The config is used to define:
   If you want greater control over your systems (to change their public access or their name), you can use the `systems` object in the config.
 - The [modules](/world/modules) that will be installed in the `World`.
 
-The is an example of a `World` config:
+This is an example of a `World` config:
 
 ```tsx
 import { defineWorld } from "@latticexyz/world";
+import { resolveTableId } from "@latticexyz/world/internal";
```
Comment (alvrs) [inline docs/pages/config.mdx:19]: We should not document things that live in `/internal`, the idea with those exports is that they're only used internally. This makes me wonder though whether we should make this util an external export since it's needed for this use case @holic?

File: docs/pages/config.mdx [hunk 7]
```diff
@@ -90,19 +99,8 @@ The global configuration keys are all optional.
   The keys are the names of the enumerations.
   The values are arrays of the strings for the values the enumeration can take.
 
-- **`tables`**: a record of tables. The keys in the record are table names.
```
Comment (alvrs) [inline docs/pages/config.mdx:93]: why was this removed?

Comment (qbzzt) [reply docs/pages/config.mdx:93]: Because in the current version we have two separate `tables` lines. [See here for the other one](https://github.com/latticexyz/mud/blob/main/docs/pages/config.mdx?plain=1#L61)

File: docs/pages/config.mdx (hunk 6 above)
Comment (holic) [reply docs/pages/config.mdx:19]: I intentionally made it internal because I am not happy with this API and want to replace it without it being considered a breaking change

**Comment by qbzzt** [comment]
The config file documentation changed so much this is no longer relevant as is.
